etw/logrus/hook.go

```diff
@@ -0,0 +1,78 @@
+package hook
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/Microsoft/go-winio/etw"
+	"github.com/sirupsen/logrus"
+
+	"golang.org/x/sys/windows"
+)
+
+// Hook is a Logrus hook which logs received events to ETW.
+type Hook struct {
+	provider *etw.Provider
+}
+
+// NewHook registers a new ETW provider and returns a hook to log from it.
+func NewHook(providerName string, providerID *windows.GUID) (*Hook, error) {

```

Comment:
`*windows.GUID`? Why `*` isn't it a `[16]byte`?

Reply:
windows.GUID is defined as this, which matches the GUID struct in Windows C/C++ code:

```
type GUID struct {
	Data1 uint32
	Data2 uint16
	Data3 uint16
	Data4 [8]byte
}
```

I used a pointer because that's how I usually see GUIDs passed around in C/C++. My assumption was the cost is about the same there as in Go, so making the same decision seemed reasonable. Happy to change that though. I don't have a good heuristic for when Go code tends to use pointers or not to pass structs around.

Reply:
Oh wow. I didn't realize `x/sys/windows` had yet another `GUID` structure. Should we take a different public structure so that we don't have to force every caller to use this and include the `x/sys/windows` package to do so?

Reply:
If I can get the GUID generation from provider name working, I think we can just remove the GUID parameter from NewProvider (and add something like NewProviderWithID that has it). Then the majority of consumers could get away with not needing `x/sys/windows` as they wouldn't need to pass a GUID.